app/assets/stylesheets/partials/authoring/modal_edit.scss

```diff
@@ -21,6 +21,18 @@
       height: 100px;
     }
 
+    &.edit_embeddable_xhtml {
+      .textblock-heading-field {
+        display: none;
```

Comment:
I think it'd be better if this didn't completely hide the heading filed, but just disabled it. That way authors would realize they could add a heading by checking the box.